packages/react-aria-components/test/DropZone.test.js

```diff
@@ -0,0 +1,455 @@
+    let dropzone = getByText('Test');
+
+    expect(dropzone).not.toHaveAttribute('data-hovered');
+    expect(dropzone).not.toHaveClass('hover');
+
+    userEvent.hover(dropzone);
+    expect(dropzone).toHaveAttribute('data-hovered', 'true');
+    expect(dropzone).toHaveClass('hover');
+
+    userEvent.unhover(dropzone);
+    expect(dropzone).not.toHaveAttribute('data-hovered');
+    expect(dropzone).not.toHaveClass('hover');
+  });
+
+  it('should support focus ring with no file input', () => {
+    let {getByTestId, getByRole} = render(
+      <DropZone data-testid="foo" className={({isFocusVisible}) => isFocusVisible ? 'focus' : ''}>
+        <Text slot="heading">
+          Test
+        </Text>
+      </DropZone>);
+    let dropzone = getByTestId('foo');
+    let button = getByRole('button');
+    
+    expect(dropzone).not.toHaveAttribute('data-focus-visible');
+    expect(dropzone).not.toHaveClass('focus');
+
+    userEvent.tab();
+    expect(document.activeElement).toBe(button);
+    expect(dropzone).toHaveAttribute('data-focus-visible', 'true');
+    expect(dropzone).toHaveClass('focus');
+
+    userEvent.tab();
+    expect(dropzone).not.toHaveAttribute('data-focus-visible');
+    expect(dropzone).not.toHaveClass('focus');
+  });
+
+  it('should apply correct aria-labelledby with no input, button, or link', () => {
+    let {getByRole, getByText} = render(
+      <DropZone className="test">
+        <Text slot="heading">
+          Test
+        </Text>
+      </DropZone>);
+    let text = getByText('Test');
+    let button = getByRole('button');
+    expect(button).toHaveAttribute('aria-labelledby', `${text.id}`);
+  });
+
+  it('should have correct aria-label with input', () => {
+    let {getByRole} = render(
+      <DropZone>
+        <Input />
+        <Text slot="heading">
+          Test
+        </Text>
+      </DropZone>);
+    let button = getByRole('button');
+    expect(button).toHaveAttribute('aria-label', 'Press enter to select a file');
+  });
+
+  it('should support render props', () => {
+    let {getByTestId} = render(
+      <DropZone data-testid="foo">
+        {({isFocused}) => isFocused ? 'Focused' : 'Not Focused'}
+      </DropZone>
+    );
+    let dropzone = getByTestId('foo');
+    
+    expect(dropzone).toHaveTextContent('Not Focused');
+    
+    userEvent.tab();
+    expect(dropzone).toHaveTextContent('Focused');
+  });
+
+  it('should support focus ring around visible button', () => {
+    let {getByTestId, getByRole} = render(
+      <DropZone data-testid="foo">
+        <Button slot="file">Upload</Button>
+        <Input data-testid="boo" />
+      </DropZone>);
+
+    let dropzone = getByTestId('foo');
+    let button = getByRole('button');
+    expect(button).toHaveAttribute('class', 'react-aria-Button');
+
+    expect(dropzone).not.toHaveAttribute('data-focus-visible');
+    expect(dropzone).not.toHaveClass('focus');
```

Comment:
is this a class we actually have? i thought we only applied a data attribute? we only seem to be checking for the negative of it, I don't see anywhere this exists

Reply:
yeah that's my bad. we can remove those checks